packages/hooks/src/useGetPhoneNumberOTP.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import * as Yup from 'yup';
+import { ValidationConstants } from '@deriv-com/utils';
+import { useMutation } from '@deriv/api';
+import { VERIFICATION_SERVICES } from '@deriv/shared';
+import { Localize, localize } from '@deriv/translations';
+
+type THandleError = {
+    code: string;
+    message: string;
+};
+
+/** A hook for requesting OTP which is sent on whatsapp or sms platforms */
+const useGetPhoneNumberOTP = () => {
+    const { data, mutate, ...rest } = useMutation('phone_number_challenge');
+    const [error_message, setErrorMessage] = React.useState<React.ReactNode>('');
+
+    const phoneNumberSchema = Yup.string().matches(
+        ValidationConstants.patterns.phoneNumber,
+        localize('Please enter a valid phone number.')
+    );
+
+    const validatePhoneNumber = (phone_number: string) => {
+        phoneNumberSchema
+            .validate(phone_number)
+            .then(() => setErrorMessage(''))
+            .catch(({ errors }: any) => {
+                setErrorMessage(errors);
+            });
+    };
```

Comment:
isnt it better to move these general phone number validations out of this request  hook?